dynamometer-workload/src/main/java/com/linkedin/dynamometer/workloadgenerator/audit/AuditReplayThread.java

```diff
@@ -49,27 +50,29 @@
   private static final Log LOG = LogFactory.getLog(AuditReplayThread.class);
 
   private DelayQueue<AuditReplayCommand> commandQueue;
+  private ConcurrentMap<String, FileSystem> fsCache;
+  private URI namenodeUri;
+  private UserGroupInformation loginUser;
+  private Configuration mapperConf;
   // If any exception is encountered it will be stored here
   private Exception exception;
   private long startTimestampMs;
-  private FileSystem fs;
-  private DFSClient dfsClient;
   private boolean createBlocks;
 
   // Counters are not thread-safe so we store a local mapping in our thread
   // and merge them all together at the end.
   private Map<REPLAYCOUNTERS, Counter> replayCountersMap = new HashMap<>();
   private Map<String, Counter> individualCommandsMap = new HashMap<>();
 
-  AuditReplayThread(Mapper.Context mapperContext) throws IOException {
+  AuditReplayThread(Mapper.Context mapperContext, ConcurrentMap<String, FileSystem> fsCache) throws IOException {
     commandQueue = new DelayQueue<>();
-    Configuration mapperConf = mapperContext.getConfiguration();
-    String namenodeURI = mapperConf.get(WorkloadDriver.NN_URI);
+    this.fsCache = fsCache;
```

Comment:
nit: remove 'this.' to be consistent with the rest?

Reply:
The `this` is necessary to quality `fsCache` as a field since `fsCache` is also the name of a parameter.